File: java-automation/src/test/java/com/myecommerce/automation/dsl/domain/DeliveryState.java
package com.myecommerce.automation.dsl.domain;

import lombok.Builder;

import java.util.List;

@Builder
public record DeliveryState(
    boolean sectionVisible,
    List<DeliveryOption> options,
    String headerText,
    boolean minimumOrderTextPresent
) {
    public long selectedOptionCount() {
        return options.stream().filter(DeliveryOption::selected).count();
    }
}

File: kotlin-automation/src/test/kotlin/com/myecommerce/automation/dsl/domain/DeliveryState.kt
package com.myecommerce.automation.dsl.domain

data class DeliveryState(
    val sectionVisible: Boolean,
    val options: List<DeliveryOption>,
    val headerText: String,
    val minimumOrderTextPresent: Boolean,
) {
    fun selectedOptionCount(): Long = options.count { it.selected }.toLong()
}
